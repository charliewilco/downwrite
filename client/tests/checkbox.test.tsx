import * as React from "react";
import "jest-dom/extend-expect";
import { render, fireEvent } from "react-testing-library";
import Check from "../components/checkbox";

let { getByTestId, container } = render(
  <label data-testid="TEST_CHECKBOX_LABEL" htmlFor="check">
    <Check data-testid="TEST_CHECKBOX" id="check" />
  </label>
);

describe("<Check />", () => {
  it("checks", () => {
    expect(getByTestId("TEST_CHECKBOX")).toBeInTheDocument();
    fireEvent.click(getByTestId("TEST_CHECKBOX_LABEL"));
    expect((getByTestId("TEST_CHECKBOX") as HTMLInputElement).checked).toBe(true);
  });

  it("matches snapshot", () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});
