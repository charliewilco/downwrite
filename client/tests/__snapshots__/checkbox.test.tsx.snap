// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Check /> matches snapshot 1`] = `
.c0 {
  display: inline-block;
  border-radius: 4px;
  vertical-align: middle;
  width: 20px;
  height: 20px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0px;
  background: #D0D0D0;
}

.c0:focus {
  outline: 1px solid;
}

<label
  data-testid="TEST_CHECKBOX_LABEL"
  for="check"
>
  <input
    class="c0"
    data-testid="TEST_CHECKBOX"
    id="check"
    type="checkbox"
  />
</label>
`;
